### ISSUE TITLE
Attempt to overlay sidecar-injector config causes yaml scalar symbol be eaten

### ISSUE THREAD

**Issue by rtmie** [istio/istio#28100]
**Bug description**

I am having an issue with using overlays in my profile and having them correctly processed by the operator/istioctl.
Specifically I want to overlay a change in the config section of the sidecar-injector.
The config section of the sidecar-injector is yaml literal scalar data which will go into the istio-sidecar-injector configmap so as a scalar my only option is to include the whole scalar in the overlay patch. 
However the scalar data itself contains valid yaml.

If I create a profile (e.g based on `istioctl dump profile default`) and add the overlay as below
```
   pilot:
      k8s:
        overlays:
        - kind: ConfigMap
          name: istio-sidecar-injector
          patches:
          - path: data.config
            value: |-
              alwaysInjectSelector: []
              injectedAnnotations: {}
              neverInjectSelector: []
              policy: enabled
              template: |
                rewriteAppHTTPProbe: {{ valueOrDefault .Values.sidecarInjectorWebhook.rewriteAppHTTPProbe false }}
                initContainers:
.
.
.
```

Then when I generate a manifest from this the literal scalar symbol in the `value: |-` section disappears:

```
apiVersion: v1
data:
  config:
    alwaysInjectSelector: []
    injectedAnnotations: {}
    neverInjectSelector: []
    policy: enabled
    template: |
      rewriteAppHTTPProbe: {{ valueOrDefault .Values.sidecarInjectorWebhook.rewriteAppHTTPProbe false }}
      initContainers:
```

As an aside I notice that if I put something into the overlaid scalar to stop it being valid yaml inside the scalar, then the manifest generation recognises the overlay as a scalar and leaves the literal scalar symbol in place.

So it seems that the overlay process interprets patch data as yaml if it contains valid yaml even though it is signalled as literal scalar
  

[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Overlay should replace the scalar data by patching `data.config` in the sidecar injector configmap within the manifest with the scalar data provided int he overlay 
**Steps to reproduce the bug**
1. generate a profile dump `istioctl profile dump default >my_profile.yaml`
2. edit my_profile.yaml to add an overlay in the `pilot:` section like
```
   pilot:
      enabled: true
      k8s:
        overlays:
        - kind: ConfigMap
          name: istio-sidecar-injector
          patches:
          - path: data.config
            value: |-
```
3. generate a manifest by: `istioctl manifest generate -f my_profile.yaml >manifest.yaml`
4. view the manifest.yaml file and observe that the overlay has produced:
```
apiVersion: v1
data:
  config:
    alwaysInjectSelector: []
    injectedAnnotations: {}
    neverInjectSelector: []
    policy: enabled
    template: |
      rewriteAppHTTPProbe: {{ valueOrDefault .Values.sidecarInjectorWebhook.rewriteAppHTTPProbe false }}
      initContainers:
      {{ if ne (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `NONE` }}
      {{ if .Values.istio_cni.enabled -}}
      - name: istio-validation
```
**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
```
$ istioctl version  --remote
client version: 1.7.1
control plane version: 1.7.1
data plane version: 1.7.1 (25 proxies)
```
```
$ kubectl version --short
Client Version: v1.19.2
Server Version: v1.18.8
```

**How was Istio installed?**
Operator 
**Environment where bug was observed (cloud vendor, OS, etc)**
Kubeadm installed cluster on ubuntu-18.04

Modified profile adding an env var to data.config to show the issue:

[my_profile.yaml.txt](https://github.com/istio/istio/files/5407229/my_profile.yaml.txt)

**Comment by howardjohn** [comment]
I suspect this may be simialar to https://github.com/istio/istio/pull/24348 which I was unable to get passing tests

**Comment by rtmie** [comment]
I think this would be resolved if [#28038](https://github.com/istio/istio/issues/28038) is implemented

**Comment by gonzaloserrano** [comment]
For the record I've been hit by the same issue and could not find a fix.

Related code, with `toYaml` usage: https://github.com/istio/istio/blob/master/manifests/charts/istiod-remote/templates/istiod-injector-configmap.yaml#L71-L73

`toYaml` non-fixed issue at the helm repo https://github.com/helm/helm/issues/4262
